### ISSUE TITLE
Huge difference in training with Pytorch backend

### ISSUE THREAD

**Issue by invoxiaglo** [keras-team/keras#20713]
I ran the simple mnist training following code with the different backends: Tensorflow, Pytorch and Jax. I get similar results with tensorflow and Jax: between 98 and 99% test accuracy but way lower results with Pytorch: below 90%. 

```python
import os
from time import time

os.environ["KERAS_BACKEND"] = "jax"
# os.environ["KERAS_BACKEND"] = "torch"
# os.environ["PYTORCH_ENABLE_MPS_FALLBACK"] = "1"

import keras
from keras import layers, Model

print(f"Keras version: {keras.__version__}")

# Load MNIST dataset
from keras.datasets import mnist
(x_train, y_train), (x_test, y_test) = mnist.load_data()

print(f"x_train shape: {x_train.shape}")
print(f"y_train shape: {y_train.shape}")

# Preprocess data
x_train = x_train.astype("float32") / 255.0
x_test = x_test.astype("float32") / 255.0
x_train = x_train[..., None]  # Add channel dimension
x_test = x_test[..., None]

# Define the model using Functional API
inputs = keras.Input(shape=(28, 28, 1), name="input_layer")
x = layers.Conv2D(32, kernel_size=(3, 3), activation="relu", name="conv_1")(inputs)
x = layers.MaxPooling2D(pool_size=(2, 2), name="pool_1")(x)
x = layers.Conv2D(64, kernel_size=(3, 3), activation="relu", name="conv_2")(x)
x = layers.MaxPooling2D(pool_size=(2, 2), name="pool_2")(x)
x = layers.Flatten(name="flatten")(x)
x = layers.Dense(128, activation="relu", name="dense_1")(x)
x = layers.Dropout(0.2, name="dropout_1")(x)
outputs = layers.Dense(10, activation="softmax", name="output_layer")(x)

model = Model(inputs=inputs, outputs=outputs, name="mnist_cnn_model")

model.summary()

# Compile the model
model.compile(optimizer="adam", 
              loss="sparse_categorical_crossentropy", 
              metrics=["accuracy"])

begin_time = time()

# Train the model
model.fit(x_train, y_train, epochs=3, batch_size=32, validation_split=0.2)

end_time = time()
print(f"Time taken: {end_time - begin_time:.2f}s")

# Evaluate the model
test_loss, test_acc = model.evaluate(x_test, y_test, verbose=2)
print(f"Test accuracy: {test_acc:.4f}")
```

Note: I have to put PYTORCH_ENABLE_MPS_FALLBACK to True for the code to run with Pytorch (I am on MAC OS). However, it is around 6 times slower than with Tensorflow or Jax.

Are there differences with the layers in Pytorch or with the optimizer (default parameters, ...) which can explain this difference in accuracy? How to fix the following code so that it gives the same results in Pytorch?

**Comment by sonali-kumari1** [comment]
Hi @invoxiaglo,

Thanks for reporting this issue. I have tested your code with macOS colab and code works fine individually with all the backends. I am getting accuracy between 98 and 99% for all the backends as well.
I am able to run the code without `os.environ["PYTORCH_ENABLE_MPS_FALLBACK"] = "1"` and time taken for all backends is pretty much the same.
Attaching [gist](https://colab.sandbox.google.com/gist/sonali-kumari1/687fce9f95d7f9619a5d6785515f82a1/20713.ipynb#scrollTo=6vvbMMlC0QLx) for your reference.

**Comment by invoxiaglo** [comment]
Thanks for your tests. I have the same results as you when I test on Colab. I attached an image of the results I get on my computer. It must be due to the old version of Pytorch I am using (2.2.2). 
<img width="525" alt="image" src="https://github.com/user-attachments/assets/a8a9c164-9362-4131-9755-68ba2103ed39" />
